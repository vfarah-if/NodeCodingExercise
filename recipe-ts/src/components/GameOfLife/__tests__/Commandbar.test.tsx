import { render, screen } from '@testing-library/react';

import {
  DefaultCommandbar,
  CustomCommandbar,
} from '../Commandbar/Commandbar.stories';

describe('Commandbar', () => {
  const generate = jest.fn();
  const clear = jest.fn();
  const randomise = jest.fn();
  const simulate = jest.fn();

  beforeEach(() => {
    generate.mockClear();
    clear.mockClear();
    randomise.mockClear();
    simulate.mockClear();
  });

  test('should default expected command settings', () => {
    const { container } = render(
      <DefaultCommandbar
        {...DefaultCommandbar.args}
        onClear={clear}
        onGenerate={generate}
        onRandomise={randomise}
        onSimulate={simulate}
      />
    );

    expect(container).toMatchSnapshot();

    //TODO : Figure out why this is not working as expected
    // const generateButton = getAndAssertButton('Generate');
    // generateButton.click();
    // expect(generate).toHaveBeenCalledTimes(1);
  });

  test('should customise commandbar settings', () => {
    const { container } = render(
      <CustomCommandbar
        {...CustomCommandbar.args}
        onClear={clear}
        onGenerate={generate}
        onRandomise={randomise}
        onSimulate={simulate}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

export function getAndAssertButton(name: string): HTMLButtonElement {
  const button = screen.getByRole('button', { name }) as HTMLButtonElement;
  expect(button).toHaveTextContent(name);
  return button;
}
