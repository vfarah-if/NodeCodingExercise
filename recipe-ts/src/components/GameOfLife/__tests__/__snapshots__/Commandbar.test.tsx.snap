// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commandbar should customise commandbar settings 1`] = `
<div>
  <header>
    <div
      class="wrapper"
    >
      <div>
        <svg
          height="58"
          version="1.0"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <rect
              fill="black"
              height="14"
              id="alive"
              stroke="#c0c0c0"
              stroke-width="2"
              width="14"
              x="0"
              y="0"
            />
            <rect
              fill="white"
              height="14"
              id="dead"
              stroke="#c0c0c0"
              stroke-width="2"
              width="14"
              x="0"
              y="0"
            />
          </defs>
          <g
            transform="translate(1,1)"
          >
            <use
              transform="translate(0,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(14,0)"
              xlink:href="#alive"
            />
            <use
              transform="translate(28,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,0)"
              xlink:href="#alive"
            />
            <use
              transform="translate(0,14)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(28,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(0,28)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(28,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,28)"
              xlink:href="#alive"
            />
            <use
              transform="translate(0,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(28,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(42,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(56,42)"
              xlink:href="#dead"
            />
          </g>
        </svg>
        <h1>
          Conway's Game of life
        </h1>
      </div>
      <div>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Make it so
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Animate
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Shuffle it
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Erase
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`Commandbar should default expected command settings 1`] = `
<div>
  <header>
    <div
      class="wrapper"
    >
      <div>
        <svg
          height="58"
          version="1.0"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <rect
              fill="black"
              height="14"
              id="alive"
              stroke="#c0c0c0"
              stroke-width="2"
              width="14"
              x="0"
              y="0"
            />
            <rect
              fill="white"
              height="14"
              id="dead"
              stroke="#c0c0c0"
              stroke-width="2"
              width="14"
              x="0"
              y="0"
            />
          </defs>
          <g
            transform="translate(1,1)"
          >
            <use
              transform="translate(0,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(14,0)"
              xlink:href="#alive"
            />
            <use
              transform="translate(28,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,0)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,0)"
              xlink:href="#alive"
            />
            <use
              transform="translate(0,14)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(28,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,14)"
              xlink:href="#dead"
            />
            <use
              transform="translate(0,28)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(28,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(42,28)"
              xlink:href="#dead"
            />
            <use
              transform="translate(56,28)"
              xlink:href="#alive"
            />
            <use
              transform="translate(0,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(14,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(28,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(42,42)"
              xlink:href="#alive"
            />
            <use
              transform="translate(56,42)"
              xlink:href="#dead"
            />
          </g>
        </svg>
        <h1>
          Game of Life on Mars
        </h1>
      </div>
      <div>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Generate
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Simulate
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Randomise
        </button>
        <button
          class="storybook-button storybook-button--small storybook-button--primary"
          type="button"
        >
          Clear
        </button>
      </div>
    </div>
  </header>
</div>
`;
